### PR TITLE
feat: sectionList

### PR DESCRIPTION
**Pull request by JakeBalakrishna**

### CODE REVIEW

File: src/components/Meals/MealList/index.tsx [hunk 1]
```diff
@@ -1,24 +1,70 @@
...
 import { MealItem } from "../MealItem";
 import { useAppSelector } from "../../../redux/store";
+import { SectionList, StyleSheet, Text } from "react-native";
+import { format } from "date-fns";
+//TODO: TS is confusing come back to this to think about it.
+type SectionObject = {
+  [formattedDate: string]: Meal[];
+};
+
+type Meal = {
```
Comment (mattwilson02) [inline src/components/Meals/MealList/index.tsx:11]: you dont need to redeclare this type, just import from Meal slice

File: src/components/Meals/MealList/index.tsx [hunk 2]
```diff
@@ -1,24 +1,70 @@
...
 
 export const MealList = () => {
   const meals = useAppSelector(state => state.meals);
 
-  //TODO: implement sectionList
+  const groupedMeals = meals.reduce((sectionObject: SectionObject, meal) => {
+    const date = meal.dateTime;
+    const formattedDate = format(date, "dd.MM.yyyy");
+
+    // If the formatted date is not a key in the sectionObject (yes its a bad name,
```
Comment (mattwilson02) [inline src/components/Meals/MealList/index.tsx:34]: i mean fair enough, its a perfect implementation, you can remove the comments

**Review by mattwilson02 — APPROVED** [review]

File: src/components/Meals/MealList/index.tsx (hunk 2 above)
Comment (JakeBalakrishna) [reply src/components/Meals/MealList/index.tsx:34]: i'll remove the part about it being a bad name but I want to leave the comment for reference in the future.

File: src/components/Meals/MealList/index.tsx (hunk 1 above)
Comment (mattwilson02) [reply src/components/Meals/MealList/index.tsx]: this hasnt been resolved yet

**Pull request merged**